python/tests/test_tables.py

```diff
@@ -2266,3 +2266,133 @@ def test_set_columns_not_implemented(self):
+            k for k, n in enumerate(tables.nodes) if n.flags & tskit.NODE_IS_SAMPLE
+        ]
+        nodesg = [node_map[n] for n in nodes]
+        # simplifying with filter_populations=False bc SLiM can start
+        # with id 1 instead of zero
+        tables.simplify(nodes, filter_populations=False)
+        tablesg.simplify(nodesg, filter_populations=False)
+        # TODO: a test for provenances
+        tables.provenances.clear()
+        tablesg.provenances.clear()
+        self.assertEqual(len(tables.nodes), len(tablesg.nodes))
+        # simplify does not put individuals in order
+        # but it does for nodes. that is why we check
+        # for equality of individuals as follows
+        for j in range(tables.nodes.num_rows):
+            na = tables.nodes[j]
+            nb = tablesg.nodes[j]
+            if na.individual != -1 or nb.individual != -1:
+                self.assertFalse((na.individual == -1) or (nb.individual == -1))
+                ia = tables.individuals[na.individual]
+                ib = tablesg.individuals[nb.individual]
+                self.assertEqual(ia.flags, ib.flags)
+                self.assertEqual(ia.metadata, ib.metadata)
+                self.assertTrue((ia.location == ib.location).all())
+            if na.population != -1 or nb.population != -1:
+                self.assertFalse((na.population == -1) or (nb.population == -1))
+                pa = tables.populations[na.population]
+                pb = tablesg.populations[nb.population]
+                self.assertEqual(pa, pb)
+            self.assertEqual(na.time, nb.time)
+            self.assertEqual(na.metadata, nb.metadata)
+            self.assertEqual(na.flags, nb.flags)
+        tables.individuals.clear()
+        tablesg.individuals.clear()
+        tables.populations.clear()
+        tablesg.populations.clear()
+        tables.nodes.clear()
+        tablesg.nodes.clear()
+        self.assertEqual(tables, tablesg)
+
+    def verify_node_populations(self, tables1, tables2, node_map):
+        tablesg, (node_map2new, pop_map2new, ind_map2new) = tables1.graft(tables2, node_map)
+        # check that ts1 pops remain unchanged
+        for j, p in enumerate(tables1.populations):
+            self.assertEqual(p, tablesg.populations[j])
+        # check that each ts2 pop maps to a unique tsg pop
+        self.assertEqual(len(pop_map2new.keys()),
+                         len(set(pop_map2new.values())))
+        # check that ts2 pops are unchanged, and that newly added nodes have a
+        # new population
+        for n2 in range(tables2.nodes.num_rows):
+            p2 = tables2.nodes[n2].population
+            ng = node_map2new[n2]
+            pg = tablesg.nodes[ng].population
+            self.assertEqual(
+                tables2.populations[p2],
+                tablesg.populations[pg])
+            if n2 not in node_map:
+                self.assertGreaterEqual(pg, len(tables1.populations))
+
+    def test_msprime_example(self):
```

Comment:
This is a good start, but we'll need a lot more tests. I think we should focus on the forwards sim case, since this is the main application. So, we'll want to run some forward sims using the local WF simulator (I guess we'll need to put some stuff about populations and migrations in there - should be easy enough?)

Reply:
we have some tests with SLiM that were left out. can we do tests using SLiM? idk what the local WF simulator is.

Reply:
Here it is: https://github.com/tskit-dev/tskit/blob/master/python/tests/test_wright_fisher.py